### ISSUE TITLE
How to cite iminuit?

### ISSUE THREAD

**Issue by iamholger** [iminuit/iminuit#179]
Hi,

we are happily using iminuit in our project  (professor.hepforge.org). We would like to cite iminuit properly but so far don't know how.

Thanks,
Holger

**Comment by cdeil** [comment]
@piti118 @Dapid @energynumbers @fabianrost84 @lgeiger @omazapa You've contributed to iminuit in the past (well, @piti118 has written almost all of it). Do you have any opinion how iminuit should be cited?

I think the options are:
A) Ask people to add a link to https://github.com/iminuit/iminuit if possible
B) Put the 1.2 or a future release on https://zenodo.org/ to get a DOI
C) Someone volunteers to take the lead on writing a paper about iminuit in the near future.

I think C could be great (nice examples, benchmarks, technical details about C / Python / Cython), but I presume there's no-one that wants to put in that work in the coming months, right?

So do you think putting it on Zenodo is useful?

**Comment by fabianrost84** [comment]
C would be great but I did almost nothing and understand nothing of the code really so I do not see myself there. I recently [published](http://elifesciences.org/lookup/doi/10.7554/eLife.10230.001) and cited iminuit using A. I don't know Zenodo and don't think a DOI is absolutely necessary. A is also [what scipy does](https://scipy.org/citing.html).

**Comment by iamholger** [comment]
I'd suggest Option B as a short term solution, it's really quick to do and since zenodo is hosted at CERN there will be long term support.

**Comment by piti118** [comment]
Would be nice if someone set up Zenodo.

**Comment by fabianrost84** [comment]
Now that I started to use Zenodo myself I changed my mind. Zenodo would be great.

**Comment by Dapid** [comment]
Another alternative for the paper: http://joss.theoj.org/

**Comment by cdeil** [comment]
+1 for http://joss.theoj.org/ and to do a short writeup of what MINUIT iminuit is.

I remember that when I learned about MINUIT and iminuit I was super confused about:
- What algorithm MINUIT implements (http://iminuit.readthedocs.io/en/latest/#technical-stuff has good references)
- What version of MINUIT is the current one and used by iminuit
- How iminuit is implemented (the `Minuit` class is mostly pure-Python, but all located in a Cython file, so has access to C++ / MINUIT)

I think just a 1-2 page write up with the basic info and references would be very helpful to many people that come to MINUIT / iminuit.

Is anyone interested in leading or collaborating on this in the near future (weeks / months)?

Of course there could be more, e.g. an application example and benchmarks, but probably no-one has time.

Just throwing a zip file archive of the repo up on Zenodo and people starting to cite that feels a bit like a missed opportunity to me given how nice and widely used MINUIT and iminuit is.

**Comment by Dapid** [comment]
Count me in, with limited bandwidth.

**Comment by fabianrost84** [comment]
+1 forhttp://joss.theoj.org/
I'd be happy to participate in writing. But given my very limited contributions and understanding of the code I might only act as an "internal reviewer". 

Concerning Zenodo, putting iminuit on Zenodo in parallel seems like a good idea to me. That way both, the software and the paper describing it would be citable. 

btw https://www.authorea.com/ seems like a good choice for writing this (and I always wanted to try it out)

**Comment by cdeil** [comment]
> Concerning Zenodo, putting iminuit on Zenodo in parallel seems like a good idea to me. That way both, the software and the paper describing it would be citable.

Isn't it better to have all citations focus on one reference, instead of being scattered across many references?
I.e. is it really helpful to post to Zenodo also or counter-productive?

I don't consider the aspect that Zenodo archives the code very important.
We have a git tag for versions, and the git repo is well-replicated across Github and contributors / user's machines and the iminuit source release is also on PyPI.
So it should be OK to say "I used iminuit version 1.2" when citing use of iminuit without needing a DOI specific to that release IMO.

> btw https://www.authorea.com/ seems like a good choice for writing this (and I always wanted to try it out)

I tried it once and didn't like it much.
I prefer just a Github repo and Makefile.
But if others want, yes, we can use Authorea.

Maybe let's wait for a bit to see what @piti118 as main iminuit author thinks.
@piti118 - Do you have time to take the lead on this or should one of us?

We could then make it time-based, i.e. start with a minimal 1-page writeup, and say: contributions / extensions welcome until October or November, i.e. some deadline. This should avoid getting stuck or having this work loom over someone's head that really doesn't have time besides their actual job.

**Comment by HenniOVP** [comment]
Hi all,

for my bachelor I have been using iminuit, with great results. Of course I would like to properly cite iminuit and its authors. Following this thread, I could not find any entries in either [The Journal of Open Source Software](https://joss.theoj.org/) or [zenodo](https://zenodo.org/).

As a user of iminuit it would be nice to have a temporary solution, until a final citation form is chosen. Most likely the simplest temporary solution would be something close to option A) mentioned at the very beginning of the discussion.

> I think the options are:
A) Ask people to add a link to https://github.com/iminuit/iminuit if possible
B) Put the 1.2 or a future release on https://zenodo.org/ to get a DOI
C) Someone volunteers to take the lead on writing a paper about iminuit in the near future.

What I would propose for now is a biblatex entry similar to what [scipy is using](https://scipy.org/citing.html#scipy-the-library). The following biblatex entry is the result of using the insights page for finding out who contributed and using scipy's example.


```
@online{iminuit,
	author = {Piti Ongmongkolkul (@piti118) and Christoph Deil (@cdeil) and Hans Dembinski (@HDembinski) and @Dapid and Chris Burr (@chrisburr) and Andrew (@energynumbers) and Fabian Rost (@fabianrost84) and Alex Pearce (@alexpearce) and Lukas Geiger (@lgeiger) and Omar Zapata (@omazapa)},
	title = {iminuit - MINUIT from Python},
	year = {2012--},
	url = {https://github.com/iminuit/iminuit},
	note = {[Online; accessed 2018.03.05]}
}
```

What are your thoughts on this?

**Comment by HDembinski** [comment]
Hi @HenniOVP I like your proposal, thanks for preparing a bibtex entry. :) It is better than offering nothing. @piti118 @cdeil What do you think?

**Comment by cdeil** [comment]
Maybe short-term add something to the docs with a suggestion how to acknowledge the use of iminuit. I think for now it's basically something like "This research has made use of iminuit (https://github.com/iminuit/iminuit), a Python interface to Minuit (reference)". Since iminuit is just the Python wrapper, I think suggesting to people to cite or acknowledge Minuit as well in some way would be appropriate? Is http://adsabs.harvard.edu/abs/1975CoPhC..10..343J a good reference for Minuit, or is there a better one that describes Minuit 2?

I would strongly prefer something on http://joss.theoj.org/ over just throwing an iminuit release on Zenodo. It could be as short as 1-2 pages, just summarising what iminuit is and giving the example from the front page of the docs. @HDembinski - do you have time to do this? If yes, please start in a new repo iminuit-paper.  I think author list would still go by contribution, i.e. @piti118 first? If this is happening, we should schedule a call with an open invite to define content of the writeup. Starting with a minimal 1-2 page writeup, and only then doing the call to discuss that paper core, and whether we want to extend it, would also be OK, might even be more productive.

**Comment by HDembinski** [comment]
@cdeil This plan for a JOSS paper sounds very good to me.

As an incremental improvement, I would like to add citation notice in the documentation based on @HenniOVP bibtex entry. In addition I think we should cite the original Minuit http://adsabs.harvard.edu/abs/1975CoPhC..10..343J in our README and documentation.

As far as I know, there is no more up-to-date reference for Minuit2. That is ok, because Minuit2 did not change the algorithm.

**Comment by HenniOVP** [comment]
Looking through google scholar and the [MINUIT homepage](https://seal.web.cern.ch/seal/snapshot/work-packages/mathlibs/minuit/), it seems, that the reference @cdeil gave is already the best explanation for what MINUIT is and what it does. All newer papers or references that I could find were manuals on how to use MINUIT. While those manuals always contain a short section on what MINUIT does, the explanation is normally a lot less complete than the one given in:
>http://adsabs.harvard.edu/abs/1975CoPhC..10..343J

Following up on a short-term temporary solution. The following entry would then fit the criteria?
```
@online{iminuit,
	author = {Piti Ongmongkolkul (@piti118) and Christoph Deil (@cdeil) and Hans Dembinski (@HDembinski) and @Dapid and Chris Burr (@chrisburr) and Andrew (@energynumbers) and Fabian Rost (@fabianrost84) and Alex Pearce (@alexpearce) and Lukas Geiger (@lgeiger) and Omar Zapata (@omazapa)},
	title = {iminuit - A Python interface to MINUIT \cite{1975CoPhC..10..343J}},
	year = {2012--},
	url = {https://github.com/iminuit/iminuit},
	note = {[Online; accessed 2018.03.05]}
}
@ARTICLE{1975CoPhC..10..343J,
   author = {{James}, F. and {Roos}, M.},
    title = "{Minuit - a system for function minimization and analysis of the parameter errors and correlations}",
  journal = {Computer Physics Communications},
     year = 1975,
    month = dec,
   volume = 10,
    pages = {343-367},
      doi = {10.1016/0010-4655(75)90039-9},
   adsurl = {http://adsabs.harvard.edu/abs/1975CoPhC..10..343J},
  adsnote = {Provided by the SAO/NASA Astrophysics Data System}
}
```

The only disadvantage of this approach is, that citing a reference inside another reference forces one to compile the document two times with bibtex, see: https://tex.stackexchange.com/questions/97244/how-to-cite-another-reference-entry-in-bibtex

**Comment by HDembinski** [comment]
@HenniOVP I would advice against citing a reference inside another reference. Usually people put both citations in the text, like so
```
[...] and then we fitted our awesome model with the iminuit\cite{iminuit-ref,Minuit-ref} module.
```
Another thing I have seen is to make a multi-paper reference, where you have a bibtex entry that is manually formatted (using type misc) and then you append several references with a semi-colon, like so
```
@misc{iminuit,
    note = {<iminuit reference, manually formatted>; <Minuit reference, manually formatted>}
}

**Comment by cdeil** [comment]
@HDembinski - Thanks for adding this!

---

> The only disadvantage of this approach is, that citing a reference inside another reference forces one to compile the document two times with bibtex

I would try to avoid that, too complex.

> Another thing I have seen is to make a multi-paper reference, where you have a bibtex entry that is manually formatted (using type misc) and then you append several references with a semi-colon

I would also suggest to avoid that.

How about proposing a sentence that people can put in the acknowledgements, and there we have simple references to both?

--

> As far as I know, there is no more up-to-date reference for Minuit2. That is ok, because Minuit2 did not change the algorithm.

That doesn't seem likely to me.
:-)

But agreed, suggesting to people to cite the original Minuit paper seems the best solution.
In the sentence there could be something like "iminuit uses Minuit2, the C++ implementation of Minuit maintained in the ROOT repository".

So that was my 2 cents. I don't feel strongly about any of these points, especially if really we get a short paper on iminuit soon which is then the proper solution.

**Comment by Dapid** [comment]
Just fixing my name:

`@online{iminuit,
	author = {Piti Ongmongkolkul (@piti118) and Christoph Deil (@cdeil) and Hans Dembinski (@HDembinski) and David Men\'endez Hurtado (@Dapid) and Chris Burr (@chrisburr) and Andrew (@energynumbers) and Fabian Rost (@fabianrost84) and Alex Pearce (@alexpearce) and Lukas Geiger (@lgeiger) and Omar Zapata (@omazapa)},
	title = {iminuit - A Python interface to MINUIT \cite{1975CoPhC..10..343J}},
	year = {2012--},
	url = {https://github.com/iminuit/iminuit},
	note = {[Online; accessed 2018.03.05]}
}`

**Comment by energynumbers** [comment]
And mine too:

```
@online{iminuit,
	author = {Piti Ongmongkolkul (@piti118) and Christoph Deil (@cdeil) and Hans Dembinski (@HDembinski) and David Men\'endez Hurtado (@Dapid) and Chris Burr (@chrisburr) and Andrew ZP Smith (@energynumbers) and Fabian Rost (@fabianrost84) and Alex Pearce (@alexpearce) and Lukas Geiger (@lgeiger) and Omar Zapata (@omazapa)},
	title = {iminuit - A Python interface to MINUIT \cite{1975CoPhC..10..343J}},
	year = {2012--},
	url = {https://github.com/iminuit/iminuit},
	note = {[Online; accessed 2018.03.05]}
}
```

**Comment by cdeil** [comment]
We could also add a `CITATION` file to the repo, in addition to pointing out how to acknowledge the use of iminuit / Minuit in the docs.

Example: https://github.com/astropy/astropy/blob/master/CITATION
I think the proposal came from here: https://software-carpentry.org/blog/2013/09/introducing-citation-files.html

**Comment by HDembinski** [comment]
I am closing this. Once we have a better reference for iminuit, we will update the citation file.
